instrumentation/base/lib/opentelemetry/instrumentation.rb

```diff
@@ -7,6 +7,7 @@
 require 'opentelemetry'
 require 'opentelemetry-registry'
 require 'opentelemetry/instrumentation/base'
+require 'opentelemetry/instrumentation/metrics_patch' if defined?(OpenTelemetry::Metrics) # maybe also add Env var check?
```

Comment:
I think these two safeguards are enough for users who don’t want metrics in instrumentation or wish to create custom metrics themselves (e.g., include the metrics SDK but not use it in instrumentation).

Is the patch (e.g., prepend) only because metrics are still experimental features? I was thinking of including them in the base but not installing if `OpenTelemetry::Metrics` is missing (similar to line 25 in `metrics_patch.rb`).

Reply:
Thank you! I think that could work! Yes, the patch is just because the metrics are experimental features. I thought this approach might feel "safer" (since all our instrumentation runs through `base`), but I don't know if truly provides a benefit.

It also tries to follow the configurator patches for metrics in the SDK: 
* https://github.com/open-telemetry/opentelemetry-ruby/blob/main/metrics_sdk/lib/opentelemetry/sdk/metrics/configuration_patch.rb 
* https://github.com/open-telemetry/opentelemetry-ruby/blob/main/sdk/lib/opentelemetry/sdk/configurator.rb#L150 

I'm happy to cut out the patch and just put everything in base. [I sorta tried that with the other prototype](https://github.com/open-telemetry/opentelemetry-ruby-contrib/pull/1128/files#diff-7379de806670e7a913ebbe1d8e27945a6367cc83c94d79e55434de15cdd16732).

Reply:
Whoops, forgot to follow up here. I've changed the structure to use conditions instead of the patches.